### ISSUE TITLE
执行训练代码直接报错

### ISSUE THREAD

**Issue by MengMengLiao** [PaddlePaddle/Paddle#17811]
版本号：paddlepaddle-gpu 1.4.0.post85  
GPU型号：Tesla P40
安装的CUDA版本 8.0，cudnn 5.0

![image](https://user-images.githubusercontent.com/39786088/58804634-e5094980-8644-11e9-9f0a-66dddee227ac.png)

**Comment by sandyhouse** [comment]
能提供一些程序信息吗，或者使用export GLOG_v=3，然后运行程序看看错误在哪里。

**Comment by MengMengLiao** [comment]
def training():

    negativeFile = "/home/****/datasets/imageContextModel/preTrainingNormalFile"
    positiveFile = "/home/*****/datasets/imageContextModel/preTrainingCheatingFile"

    trainDataProcess = TrainingDataProcessing()

    multiCategoryFileList = []
    negativeFileList = trainDataProcess.ExtraAndRankingImageFiles(negativeFile)
    positiveFileList = trainDataProcess.ExtraAndRankingImageFiles(positiveFile)

    print "Length of negativeFileList is:", len(negativeFileList)
    print "Length of positiveFileList is:", len(positiveFileList)

    multiCategoryFileList.append(negativeFileList)
    multiCategoryFileList.append(positiveFileList)

    trainingFileList, testingFileList = trainDataProcess.LoadImageFiles(multiCategoryFileList, [2, 1], 0.7, 2)

    print "Length of trainingFileList is:", len(trainingFileList[0]), "---", len(trainingFileList[1])
    print "Length of testingFileList is:", len(testingFileList[0]), "---", len(testingFileList[1])

    organizedTrainingFileList = trainDataProcess.OrganizingTheTrainingData(trainingFileList, 2)
    numOfTrainingSamples = len(organizedTrainingFileList)

    organizedValidatingFileList = trainDataProcess.OrganizingTheTrainingData(testingFileList, 2)
    numOfValidatingSamples = len(organizedValidatingFileList)

    print "Length of organizedTrainingFileList:", numOfTrainingSamples
    print "Length of organizedValidatingFileList:", numOfValidatingSamples
   
   """加载训练数据"""
    
    imageHeight = 224
    imageWidth  = 224
    dim = 3
    learningRate  = 0.00001
    trainingIters = 10000
    batchSize = 32
    classes = 2
    displayStep = 10
  
 """模型参数"""

    image = fluid.layers.data(name="images", shape=[None, dim, imageHeight, imageWidth], append_batch_size=False)
    label = fluid.layers.data(name="lables", shape=[None, 1], append_batch_size=False, dtype='int64')

    resNetModel = ResNet50()
    pred = resNetModel.net(image, class_dim = classes)

    loss     = fluid.layers.mean(fluid.layers.softmax_with_cross_entropy(pred, label))
    accuracy = fluid.layers.accuracy(pred, label=label)

    adam = fluid.optimizer.Adam(learning_rate=learningRate)
    adam.minimize(loss)

    train_exe = fluid.Executor(fluid.CUDAPlace(0))

    #Initializing the params of the network
    train_exe.run(program=fluid.default_startup_program())

    main_program = fluid.default_main_program()
    test_program = main_program.clone(for_test=True)

    iters = 0
    train_writer = tf.summary.FileWriter("./ResnetModelPretraining/")
    index_m = 0
    index_n = 0
    maxAccuracy = 0
    trainingAccuracy = 0
    isFine = False
    """开始训练"""
    while iters < trainingIters:

        numOfTrainingBatches = 1
        trainingLoss = 0
        for i in range(0, numOfTrainingSamples, batchSize):

            start = i
            end = i + batchSize
            if end > numOfTrainingSamples:
                end = numOfTrainingSamples

            batchTrainingSamples = organizedTrainingFileList[start: end]
            batchTrainingData  = np.zeros(shape=[end - start, 3, imageHeight, imageWidth], dtype=np.float32)
            batchTrainingLabel = np.zeros(shape=[end - start, 1], dtype=np.int64)

            k = 0
            for j in range(len(batchTrainingSamples)):

                #print batchTrainingSamples[j][1]
                if batchTrainingSamples[j][0] == 0:

                    try:
                        image = Image.open(batchTrainingSamples[j][1])
                        imageType = image.format
                        imageMode = image.mode

                        if imageType == "JPEG" and imageMode == "RGB":

                            imageTemp = cv2.imread(batchTrainingSamples[j][1], cv2.IMREAD_UNCHANGED)
                            if len(imageTemp.shape) == 2:
                                imageTemp = cv2.cvtColor(imageTemp, cv2.COLOR_GRAY2BGR)

                            imageTemp = cv2.resize(imageTemp, (imageWidth, imageHeight), interpolation=cv2.INTER_LINEAR)
                            imageTemp = np.reshape(imageTemp, (3, imageHeight, imageWidth))

                            batchTrainingData[k, :, :] = imageTemp / 255.0
                            batchTrainingLabel[k][0] = 0
                            k += 1

                        else:

                            continue

                    except Exception as error:

                        continue

                elif batchTrainingSamples[j][0] == 1:

                    try:
                        image = Image.open(batchTrainingSamples[j][1])
                        imageType = image.format
                        imageMode = image.mode

                        if imageType == "JPEG" and imageMode == "RGB":
                            imageTemp = cv2.imread(batchTrainingSamples[j][1], cv2.IMREAD_UNCHANGED)
                            if len(imageTemp.shape) == 2:
                                imageTemp = cv2.cvtColor(imageTemp, cv2.COLOR_GRAY2BGR)

                            imageTemp = cv2.resize(imageTemp, (imageWidth, imageHeight), interpolation=cv2.INTER_LINEAR)
                            imageTemp = np.reshape(imageTemp, (3, imageHeight, imageWidth))

                            batchTrainingData[k, :, :] = imageTemp / 255.0
                            batchTrainingLabel[k][0] = 1

                            k += 1
                        else:

                            continue

                    except Exception as error:

                        continue

            print "k = ", k
            batchTrainingData  = batchTrainingData [0:k]
            batchTrainingLabel = batchTrainingLabel[0:k]

            loss_accuracy = train_exe.run(main_program, feed={"images":batchTrainingData, "lables":batchTrainingLabel}, fetch_list=[loss, accuracy])

            trainingLoss += loss_accuracy[0]

            if iters == 0 and numOfTrainingBatches == 1:

                trainingAccuracy = loss_accuracy[1]
            else:

                trainingAccuracy = 0.99 * trainingAccuracy + 0.01 * loss_accuracy[1]

            #print 'numOfTrainingBatches = ', numOfTrainingBatches
            numOfTrainingBatches += 1

            if numOfTrainingBatches % displayStep == 0 and iters >= 1:

                train_writer.add_summary(
                    tf.Summary(value=[tf.Summary.Value(tag='train_accuracy', simple_value=trainingAccuracy)]),
                    iters * (numOfTrainingSamples // batchSize) + numOfTrainingBatches)
                train_writer.add_summary(
                    tf.Summary(value=[tf.Summary.Value(tag='train_loss', simple_value=trainingLoss)]),
                    iters * (numOfTrainingSamples // batchSize) + numOfTrainingBatches)
                train_writer.flush()

                validationLoss = 0
                validationAccuracy = 0
                indexOfUncorrect = 0
                test_exe = fluid.Executor(fluid.CUDAPlace(0))
                for m in range(0, numOfValidatingSamples, batchSize * 2):

                    start = m
                    end = m + (batchSize * 2)

                    if end > numOfValidatingSamples:
                        end = numOfValidatingSamples

                    batchValidatingSamples = organizedValidatingFileList[start:end]
                    batchValidatingData = np.zeros(shape=[end - start, 3, imageHeight, imageWidth], dtype=np.float32)
                    batchValidatingLabel = np.zeros(shape=[end - start, 1], dtype=np.int64)

                    k = 0
                    for n in range(len(batchValidatingSamples)):

                        if batchValidatingSamples[n][0] == 0:

                            try:
                                image = Image.open(batchValidatingSamples[n][1])
                                imageType = image.format
                                imageMode = image.mode

                                if imageType == "JPEG" and imageMode == "RGB":
                                    imageTemp = cv2.imread(batchValidatingSamples[n][1], cv2.IMREAD_UNCHANGED)
                                    if len(imageTemp.shape) == 2:
                                        imageTemp = cv2.cvtColor(imageTemp, cv2.COLOR_GRAY2BGR)
                                    imageTemp = cv2.resize(imageTemp, (imageWidth, imageHeight),
                                                           interpolation=cv2.INTER_LINEAR)
                                    imageTemp = np.reshape(imageTemp, (3, imageHeight, imageWidth))

                                    batchValidatingData[k, :, :] = imageTemp / 255.0
                                    batchValidatingLabel[k][0] = 0
                                    k += 1
                                else:

                                    continue

                            except Exception as error:

                                continue


                        elif batchValidatingSamples[n][0] == 1:

                            try:
                                image = Image.open(batchValidatingSamples[n][1])
                                imageType = image.format
                                imageMode = image.mode

                                if imageType == "JPEG" and imageMode == "RGB":
                                    imageTemp = cv2.imread(batchValidatingSamples[n][1], cv2.IMREAD_UNCHANGED)
                                    if len(imageTemp.shape) == 2:
                                        imageTemp = cv2.cvtColor(imageTemp, cv2.COLOR_GRAY2BGR)
                                    imageTemp = cv2.resize(imageTemp, (imageWidth, imageHeight),
                                                           interpolation=cv2.INTER_LINEAR)
                                    imageTemp = np.reshape(imageTemp, (3, imageHeight, imageWidth))

                                    batchValidatingData[k, :, :] = imageTemp / 255.0
                                    batchValidatingLabel[k][0] = 1
                                    k += 1
                                else:

                                    continue

                            except Exception as error:

                                continue

                    print k
                    batchValidatingData  = batchValidatingData [0: k]
                    batchValidatingLabel = batchValidatingLabel[0: k]

                    loss_accuracy = test_exe.run(test_program, feed={'images':batchValidatingData, 'lables':batchValidatingLabel}, fetch_list=[loss, accuracy])

                    loss     = loss_accuracy[0]
                    accuracy = loss_accuracy[1]

                    if accuracy < 1.0:
                        indexOfUncorrect += 1

                    validationLoss += loss
                    if m == 0:
                        validationAccuracy = accuracy
                    else:

                        validationAccuracy = 0.9 * validationAccuracy + 0.1 * accuracy

                if numOfTrainingBatches == displayStep and iters == 1:

                    minValidationLoss = validationLoss
                    minLoss = validationLoss + trainingLoss
                    smothingValidatingLoss = validationLoss
                    smothingValidatingAccuracy = validationAccuracy
                else:

                    smothingValidatingLoss     = 0.99 * smothingValidatingLoss     + 0.01 * validationLoss
                    smothingValidatingAccuracy = 0.99 * smothingValidatingAccuracy + 0.01 * validationAccuracy

                if smothingValidatingLoss < minValidationLoss:

                    minValidationLoss = smothingValidatingLoss
                    index_m = 0
                else:

                    index_m += 1

                if (validationLoss + trainingLoss) <= minLoss:
                    minLoss = validationLoss + trainingLoss

                if (validationAccuracy > 0.990 and trainingAccuracy > 0.9990) or (minLoss == (validationLoss + trainingLoss)):

                    if validationAccuracy > 0.990 and trainingAccuracy > 0.9990 and (
                            validationLoss + trainingLoss - minLoss) <= 0.15:
                        index_n += 1
                        isFine = True

                    elif minLoss == (validationLoss + trainingLoss):

                        isFine = True

                train_writer.add_summary(tf.Summary(value=[
                    tf.Summary.Value(tag='smoothing_validation_accuracy', simple_value=smothingValidatingAccuracy)]),
                                         iters * (numOfTrainingSamples // batchSize) + numOfTrainingBatches)
                train_writer.add_summary(
                    tf.Summary(value=[tf.Summary.Value(tag='validation_loss', simple_value=validationLoss)]),
                    iters * (numOfTrainingSamples // batchSize) + numOfTrainingBatches)
                train_writer.add_summary(
                    tf.Summary(value=[tf.Summary.Value(tag='validation_accuracy', simple_value=validationAccuracy)]),
                    iters * (numOfTrainingSamples // batchSize) + numOfTrainingBatches)
                train_writer.add_summary(tf.Summary(
                    value=[tf.Summary.Value(tag='smoothing_validation_loss', simple_value=smothingValidatingLoss)]),
                                         iters * (numOfTrainingSamples // batchSize) + numOfTrainingBatches)
                train_writer.add_summary(
                    tf.Summary(value=[tf.Summary.Value(tag='loss_increase_num', simple_value=index_m)]),
                    iters * (numOfTrainingSamples // batchSize) + numOfTrainingBatches)
                train_writer.add_summary(
                    tf.Summary(value=[tf.Summary.Value(tag='validation_uncorrect_num', simple_value=indexOfUncorrect)]),
                    iters * (numOfTrainingSamples // batchSize) + numOfTrainingBatches)
                train_writer.add_summary(
                    tf.Summary(value=[tf.Summary.Value(tag='model_fine_num', simple_value=index_n)]),
                    iters * (numOfTrainingSamples // batchSize) + numOfTrainingBatches)
                train_writer.add_summary(
                    tf.Summary(value=[tf.Summary.Value(tag='min_valiation_Loss', simple_value=minLoss)]),
                    iters * (numOfTrainingSamples // batchSize) + numOfTrainingBatches)
                train_writer.flush()

                if trainingAccuracy + validationAccuracy > maxAccuracy or isFine:

                    if trainingAccuracy + validationAccuracy > maxAccuracy:
                        maxAccuracy = trainingAccuracy + validationAccuracy


                    if isFine == True:

                        fluid.io.save_persistables(train_exe, '../imageContextResnetModel', main_program)

                trainingLoss = 0

            if index_m > 50 or index_n >= 30:
                break

        if index_m > 50 or index_n >= 30:
            break

        iters += 1

"""上面的是代码，不知道哪里写错了还是怎么了，我用screen新开后台窗口训练，就直接报错"""

**Comment by sandyhouse** [comment]
你在运行前export GLOG_v=3， 然后运行程序会报什么错呢？如果没有报错信息可以试试export GLOG_v=10。

**Comment by sandyhouse** [comment]
另外，你直接import paddle.fluid有问题吗？
